improver/utilities/mathematical_operations.py

```diff
@@ -404,3 +406,236 @@ def fast_linear_fit(
+                    f"The following units were found: {variance_cube.units}, {diagnostic_cube.units}"
+                )
+        if mean_cube.units != diagnostic_cube.units:
+            errors.append(
+                f"The mean cube must have the same units as the diagnostic cube."
+                f"The following units were found: {mean_cube.units}, {diagnostic_cube.units}"
+            )
+        if errors:
+            raise ValueError("\n".join(errors))
+
+    @staticmethod
+    def verify_grids_match(diagnostic_cube, mean_cube, variance_cube=None) -> None:
+        """Check that all cubes have the same spatial coordinates (i.e. the same grid coordinates or spot index coordinates)."""
+        cubes_to_check = [mean_cube, variance_cube] if variance_cube else [mean_cube]
+        """Check that all cubes have the same spatial coordinates (i.e. the same grid)"""
+        if any("spot_index" in get_dim_coord_names(cube) for cube in cubes_to_check):
+            if any(
+                cube.coord("spot_index").points
+                != diagnostic_cube.coord("spot_index").points
+                for cube in cubes_to_check
+            ):
+                raise ValueError("The index coordinates must match.")
+        elif not spatial_coords_match(cubes_to_check):
+            raise ValueError("The spatial coordinates must match.")
+
+    def verify_time_coords_match(
+        self, diagnostic_cube, mean_cube, variance_cube=None
+    ) -> None:
+        """Check that all cubes have the same time coordinates."""
+        cubes_to_check = [mean_cube, variance_cube] if variance_cube else [mean_cube]
+        errors = []
+
+        if len(cubes_to_check) == 2:
+            if not np.array_equal(
+                cubes_to_check[0].coord("time").bounds,
+                cubes_to_check[1].coord("time").bounds,
+            ):
+                errors.append(
+                    f"The mean and variance cubes must have the same bounds. The following bounds were found:"
+                    f"mean_cube bounds: {cubes_to_check[0].coord('time').bounds}, "
+                    f"variance_cube bounds: {cubes_to_check[1].coord('time').bounds}"
+                )
+
+        if not self.ignore_temporal_mismatch:
+            if not (
+                diagnostic_cube.coord("time").cell(0).point
+                <= mean_cube.coord("time").cell(0).bound[-1]
+                and diagnostic_cube.coord("time").cell(0).point
+                >= mean_cube.coord("time").cell(0).bound[0]
+            ):
+                errors.append(
+                    f"The diagnostic cube's time points must fall within the bounds of the mean cube. The following was found:"
+                    f"diagnostic cube point: {diagnostic_cube.coord('time').cell(0).point},"
+                    f"mean cube upper bound: {mean_cube.coord('time').cell(0).bound[-1]},"
+                    f"mean cube lower bound:{mean_cube.coord('time').cell(0).bound[0]}"
+                )
```

Comment:
```suggestion
        if not self.ignore_temporal_mismatch:
        diagnostic_max = diagnostic_cube.coord("time").cell(-1)
        diagnostic_min = diagnostic_cube.coord("time").cell(0)
            if not (
                diagnostic_max
                <= mean_cube.coord("time").cell(0).bound[-1]
                diagnostic_min
                >= mean_cube.coord("time").cell(0).bound[0]
            ):
                errors.append(
                    f"The diagnostic cube's time points must fall within the bounds of the mean cube. The following was found:"
                    f"diagnostic cube maximum: {diagnostic_max},"
                    f"diagnostic cube minimum: {diagnostic_min},"
                    f"mean cube upper bound: {mean_cube.coord('time').cell(0).bound[-1]},"
                    f"mean cube lower bound:{mean_cube.coord('time').cell(0).bound[0]}"
                )
```

I think that this change will allow multi-time cubes to pass through if all of the time points on the diagnostic cube are within the bounds of the climatological values.

Reply:
Added